test/format-test.cc

```diff
@@ -2473,3 +2473,37 @@ TEST(FormatTest, FormatUTF8Precision) {
   EXPECT_EQ(result.size(), 5);
   EXPECT_EQ(from_u8str(result), from_u8str(str.substr(0, 5)));
 }
+
+#ifdef __cpp_decltype_auto
+struct lazy_optional {
+  bool engaged;
+  std::string value;
+};
```

Comment:
I'd minimize this even further into an empty struct and make the name reflect what we are trying to test here.